R/lancelot.R

```diff
@@ -1553,37 +1532,31 @@ lancelot_parameters_strain <- function(strain_transmission, strain_seed_date,
   assert_relatives(strain_transmission)
 
   if (is.null(strain_seed_date)) {
-    if (!is.null(strain_seed_rate)) {
-      stop(paste("As 'strain_seed_date' is NULL, expected 'strain_seed_rate'",
+    if (!is.null(strain_seed_size)) {
+      stop(paste("As 'strain_seed_date' is NULL, expected 'strain_seed_size'",
                  "to be NULL"))
     }
+    if (!is.null(strain_seed_pattern)) {
+      stop(paste("As 'strain_seed_date' is NULL, expected",
+                  "'strain_seed_pattern' to be NULL"))
+    }
     strain_seed_step <- 0
   } else {
     if (length(strain_transmission) == 1L) {
       stop("Can't use 'strain_seed_date' if only using one strain")
     }
-    if (length(strain_seed_date) != length(strain_seed_rate)) {
-      stop("'strain_seed_date' and 'strain_seed_rate' must be the same length")
-    }
     assert_sircovid_date(strain_seed_date)
-    assert_increasing(strain_seed_date, strict = FALSE)
-    assert_non_negative(strain_seed_rate)
+    assert_non_negative(strain_seed_size)
+    assert_positive(strain_seed_pattern)
+
+    initial_strain_seed_step <- strain_seed_date * 4
 
-    ## The + 1 here prevents the start of the next day having the
-    ## same seeding value
+    ## This gives us zeroes until the start of seeding, then the seeding period,
+    ## and finally a 0 at the end (no seeding after the seeding period)
     strain_seed_step <-
-      numeric(strain_seed_date[[length(strain_seed_date)]] / dt)
-    for (j in seq_along(strain_seed_date)) {
-      if (j == length(strain_seed_date)) {
-        i <- length(strain_seed_step)
-      } else {
-        i <- seq.int(
-          strain_seed_date[[j]] / dt,
-          strain_seed_date[[j + 1]] / dt - 1
-        )
-      }
-      strain_seed_step[i] <- strain_seed_rate[[j]] * dt
-    }
+      c(strain_seed_step <- numeric(floor(initial_strain_seed_step)),
+        strain_seed_size *
+          seed_over_steps(initial_strain_seed_step, strain_seed_pattern), 0)
```

Comment:
This zero should not be needed? Or is this not updated to use the new offset approach? Could update that later perhaps

Reply:
Updated the odin approach for seeding the strain so this is outdated